### PR TITLE
document for specifying the `buildHttp.proxy` for `HttpUrlPlugin`

### PR DESCRIPTION
**Pull request by aleen42**
For https://github.com/webpack/webpack/pull/15354

closes https://github.com/webpack/webpack.js.org/issues/5994

- [x] Read and sign the [CLA][1]. PRs that haven't signed it won't be accepted.
- [x] Make sure your PR complies with the [writer's guide][2].
- [x] Review the diff carefully as sometimes this can reveal issues.
- [x] Do not abandon your Pull Request: [Stale Pull Requests][3].
- **Remove these instructions from your PR as they are for your eyes only.**

[1]: https://github.com/openjs-foundation/EasyCLA#openjs-foundation-cla
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

File: src/content/configuration/experiments.mdx [hunk 1]
```diff
@@ -164,6 +164,14 @@ Define the location to store the lockfile.
 
 By default webpack would generate a `<compiler-name.>webpack.lock` file>. Make sure to commit it into a version control system. During the `production` build, webpack will build those modules beginning with `http(s):` protocol from the lockfile and caches under [`experiments.buildHttp.cacheLocation`](#experimentsbuildhttpcachelocation).
 
+#### experiments.buildHttp.proxy
+
+Define the proxy to specify the proxy server for remote resources.
```
Comment (chenxsan) [inline src/content/configuration/experiments.mdx:169]:
```suggestion
Specify the proxy server to use for fetching remote resources.
```

File: src/content/configuration/experiments.mdx [hunk 2]
```diff
@@ -164,6 +164,14 @@ Define the location to store the lockfile.
...
+
+- Type: `string`
+
+By default webpack would try to use the `http_proxy` environment variable to specify the proxy server for remote resources. Besides, you can define the proxy server by setting the `proxy` option.
```
Comment (chenxsan) [inline src/content/configuration/experiments.mdx:173]:
```suggestion
By default webpack would imply the proxy server to use for fetching remote resources from the `http_proxy` environment variable. However, you can also specify one through the `proxy` option.
```

Should it be `https_proxy`?

**Review by chenxsan — COMMENTED** [review]
Hi, thanks for your patience. I've left some suggestions, let me know what you think.

File: src/content/configuration/experiments.mdx (hunk 2 above)
Comment (aleen42) [reply src/content/configuration/experiments.mdx]: Confirmed that using `http_proxy`.

Comment (snitin315) [reply src/content/configuration/experiments.mdx]: Is this case sensitive? or both `http_proxy`/`HTTP_PROXY` works?

Comment (snitin315) [reply src/content/configuration/experiments.mdx]: Works for me. /cc @chenxsan

**Review by chenxsan — APPROVED** [review]

**Pull request merged**